feat: implement airport guide pages

ukeSJTU opened the pull request:
## 描述

本 PR 实现了机场攻略功能模块，为用户提供热门机场信息、机场详细介绍以及乘机流程指引。

1. 入口设置 (`/flights`)
  - 在机票首页"更多服务"菜单下添加"机场攻略"入口按钮
  - 配置路由跳转到机场攻略主页
2. 机场攻略主页 (`/flights/guide`)
  - 实现左右布局结构
  - 左侧主内容区：
    - 使用 Grid 布局展示国内/国际热门机场
    - 基于 cities 表的 is_popular 和 isDomestic 字段筛选数据
    - 机场项格式："[城市名] [机场名]"
    - 点击跳转到对应机场详情页
  -  右侧功能区：
    - 上方："乘机流程"跳转按钮
    - 下方：天气卡片（使用 mock 静态数据）
3. 机场详情页 (`/flights/guide/airport-[iataCode]`)
  - 使用动态路由，基于机场 IATA 代码
  - 实现 Tabs 组件，包含四个标签页：
    - 机场首页
    - 机场简介
    - 机场交通
    - 机场电话
  - 客户端生成与机场相关的随机介绍内容
4. 乘机流程页 (`/flights/guide/process`)
  - 展示 4 张固定的乘机流程图片
  - 图片资源存放于 `public/flights-process/` 目录

## 相关问题

Resolves #224 

## 变更类型

- [ ] Bug 修复（不会破坏现有功能的非破坏性变更）
- [x] 新功能（添加功能的非破坏性变更）
- [ ] 破坏性变更（会导致现有功能无法正常工作的修复或功能）
- [ ] 文档更新
- [ ] 性能改进
- [ ] 代码重构
- [ ] CI/CD 改进

## 测试

- [ ] 我已经在本地测试了这些更改
- [ ] 我已经添加了证明我的修复有效或我的功能工作的测试
- [x] 新的和现有的单元测试在我的更改下都能通过
- [x] 我已经检查了代码能够无错误地构建

## 截图（如果适用）

下面这些截图对应的数据场景为 `pnpm db:seed --scenario=demo`:

<img width="1080" height="1141" alt="image" src="https://github.com/user-attachments/assets/a6f1b5c9-2099-4e10-b3ee-2a606e5a96e3" />

<img width="1080" height="1141" alt="image" src="https://github.com/user-attachments/assets/2e23f088-24f9-41a9-931f-0930a51d20e7" />

<img width="1080" height="1141" alt="image" src="https://github.com/user-attachments/assets/70630728-bd75-407c-bd76-7541ce9b8d4d" />

<img width="1080" height="1141" alt="image" src="https://github.com/user-attachments/assets/d39e3010-e99d-48a1-b5bd-80fb66eec61b" />

<img width="1080" height="1141" alt="image" src="https://github.com/user-attachments/assets/43251446-3507-4ae7-8e7c-5f321ec7059c" />

## 检查清单

- [x] 我的代码遵循此项目的代码风格
- [x] 我已经对自己的代码进行了自我审查
- [x] 我已经对代码进行了注释，特别是在难以理解的地方
- [x] 我已经对文档进行了相应的更改
- [x] 我的更改没有产生新的警告
- [x] 我已经添加了证明我的修复有效或我的功能工作的测试
- [x] 新的和现有的单元测试在我的更改下都能通过

## 代码审查检查清单（供审查者使用）

- [ ] 代码可读且有良好的文档
- [ ] 更改经过了适当的测试
- [ ] 没有明显的性能问题
- [ ] 已解决安全考虑
- [ ] 破坏性变更已得到适当记录

## 附加说明

按照 Issue 要求，机场详情页面的大部分文字内容为固定的静态数据，但确保机场名字是对应的。

Takagi-v committed:
feat: add airport guide feature with comprehensive documentation

ukeSJTU committed:
refactor: move flights guide pages to with-sidebar groups

ukeSJTU committed:
refactor: rename flights process pictures to better naming

ukeSJTU committed:
refactor(flights): update process images to more descriptive filenames

ukeSJTU committed:
refactor(flights): simplify process data structure and enhance breadcrumb navigation

ukeSJTU committed:
style(flights): update layout of boarding process header and description

ukeSJTU committed:
feat(cities): add relations for cities and airports in schema

ukeSJTU committed:
feat(airport-guide): add CityWithAirports interface and update getPopularAirports function

ukeSJTU committed:
feat(flights): restructure boarding process page with grouped processes and icons

ukeSJTU committed:
feat(flights): enhance services page with detailed service cards and coming soon indicators

ukeSJTU committed:
feat(airport-guide): update airport display logic and improve data mapping for cities

ukeSJTU committed:
feat(airport-guide): update weather card comment to clarify use of hardcoded data

ukeSJTU committed:
feat(airport-guide): replace breadcrumb implementation with new components for better structure

ukeSJTU committed:
feat(airport-guide): adjust grid layout for domestic and international airports display

ukeSJTU committed:
feat(airport-guide): improve layout and styling for airport display sections

ukeSJTU committed:
feat(airport-guide): add function to get airport by IATA code with city information

ukeSJTU committed:
feat(airport-guide): add action to retrieve airport by IATA code

ukeSJTU committed:
feat(airport-guide): update airport detail page to fetch airport data by IATA code

ukeSJTU committed:
feat(flights): export airport-related functions from city.repository

ukeSJTU committed:
docs(airport-guide): remove outdated airport guide feature documentation

ukeSJTU commented on app/(frontend)/(with-sidebar)/flights/guide/process/page.tsx:
```diff
@@ -0,0 +1,115 @@
+
+              {/* 流程图列表 - 垂直堆叠 */}
+              <div className="space-y-6">
+                {group.processes.map(process => (
+                  <Image
+                    src={process.src}
+                    alt={process.title}
+                    width={1200}
+                    height={300}
+                    className="w-full h-auto"
+                  />
```
Resolved

ukeSJTU commented on docs/airport-guide-feature.md:
```diff
@@ -0,0 +1,40 @@
+- **Pages**:
+    - `app/(frontend)/(without-sidebar)/flights/guide/page.tsx`: Homepage listing popular airports.
+    - `app/(frontend)/(without-sidebar)/flights/guide/[slug]/page.tsx`: Detail page for specific airports (e.g., `/flights/guide/airport-BJS`).
+    - `app/(frontend)/(without-sidebar)/flights/guide/process/page.tsx`: Static boarding process page.
```
Document has been marked as outdated and deleted

ukeSJTU commented on app/(frontend)/(with-sidebar)/flights/guide/[slug]/page.tsx:
```diff
@@ -0,0 +1,314 @@
+
+interface PageProps {
+  params: Promise<{
+    slug: string;
+  }>;
+}
+
+export default async function AirportDetailPage({ params }: PageProps) {
+  const { slug } = await params;
+
+  // Extract IATA code from slug (e.g., "airport-PEK" -> "PEK")
```
In Next.js v15, slug params should be decalred as Promise as described here: https://nextjs.org/docs/15/app/api-reference/file-conventions/dynamic-routes

ukeSJTU committed:
feat(boarding-process): add key prop to image elements in process mapping

Wilfrid-Hervey commented:
1. app/_actions/airport-guide.ts的逻辑用server component或许会更好
2. server action调用后端的repository和service，可以分离一下，不直接调用repository，如果之后没太多新的功能不分离也行

Wilfrid-Hervey commented:
/gemini review

ukeSJTU commented:
> 1. app/_actions/airport-guide.ts的逻辑用server component或许会更好
> 2. server action调用后端的repository和service，可以分离一下，不直接调用repository，如果之后没太多新的功能不分离也行

重构后的框架有没有文档说明？比如说要在页面上获取数据/更新数据，是谁调用谁的关系？

ukeSJTU committed:
feat(flights/guide): extract guide components and refactor guide pages

Add AirportList, GuideSidebar, BoardingProcessCard, WeatherCard, and loading skeletons.

Update pages to use new components and export airport-guide actions.
